{"uid":"434287bc113e8158","name":"TC 10.02.01 - Verify 'MEN'S DEALS' title is correct","fullName":"tests.test_sale.TestSideMenuBarMen#test_tc_10_02_01","historyId":"7d9cc06b830a248f9ded0f57b7152948","time":{"start":1703038493571,"stop":1703038498659,"duration":5088},"description":"Check title of Men's Deals section","descriptionHtml":"<p>Check title of Men's Deals section</p>\n","status":"skipped","statusMessage":"XFAIL This test is expecting failure == Assertion Error because of error in spelling Men's\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5588093e3d33 <unknown>\n#1 0x5588090a0f87 <unknown>\n#2 0x5588090f0c0e <unknown>\n#3 0x5588090f0d11 <unknown>\n#4 0x55880913c004 <unknown>\n#5 0x55880911a36d <unknown>\n#6 0x558809139297 <unknown>\n#7 0x55880911a0e3 <unknown>\n#8 0x5588090e2044 <unknown>\n#9 0x5588090e344e <unknown>\n#10 0x5588093a8861 <unknown>\n#11 0x5588093ac785 <unknown>\n#12 0x558809396285 <unknown>\n#13 0x5588093ad41f <unknown>\n#14 0x55880937a20f <unknown>\n#15 0x5588093d1028 <unknown>\n#16 0x5588093d11f7 <unknown>\n#17 0x5588093e2ed4 <unknown>\n#18 0x7fb6ad294ac3 <unknown>","statusTrace":"self = <test_sale.TestSalePage.TestSideMenuBarMen object at 0x7f3c099589d0>\nsale_page = <pages.sale_page.SalePage object at 0x7f3c095ff890>\n\n    @allure.title(\"TC 10.02.01 - Verify 'MEN'S DEALS' title is correct\")\n    @pytest.mark.xfail(reason=\"This test is expecting failure == Assertion Error \"\n                              \"because of error in spelling Men's\")\n    def test_tc_10_02_01(self, sale_page):\n        \"\"\"Check title of Men's Deals section\"\"\"\n>       title = sale_page.check_text_in_men_deals_title()\n\ntests/test_sale.py:65: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/sale_page.py:19: in check_text_in_men_deals_title\n    men_deals_title = self.element_is_visible(self.side_bar_locators.MEN_DEALS_TITLE)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"747ce82830cbff6b829f03a4515041ba\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7f3c09b16c00>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5588093e3d33 <unknown>\nE       #1 0x5588090a0f87 <unknown>\nE       #2 0x5588090f0c0e <unknown>\nE       #3 0x5588090f0d11 <unknown>\nE       #4 0x55880913c004 <unknown>\nE       #5 0x55880911a36d <unknown>\nE       #6 0x558809139297 <unknown>\nE       #7 0x55880911a0e3 <unknown>\nE       #8 0x5588090e2044 <unknown>\nE       #9 0x5588090e344e <unknown>\nE       #10 0x5588093a8861 <unknown>\nE       #11 0x5588093ac785 <unknown>\nE       #12 0x558809396285 <unknown>\nE       #13 0x5588093ad41f <unknown>\nE       #14 0x55880937a20f <unknown>\nE       #15 0x5588093d1028 <unknown>\nE       #16 0x5588093d11f7 <unknown>\nE       #17 0x5588093e2ed4 <unknown>\nE       #18 0x7fb6ad294ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1703034003307,"stop":1703034003370,"duration":63},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"clear_allure_result_folder","time":{"start":1703034003370,"stop":1703034003370,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"driver","time":{"start":1703038486092,"stop":1703038487129,"duration":1037},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"sale_page","time":{"start":1703038487129,"stop":1703038493571,"duration":6442},"status":"passed","steps":[{"name":"Open a browser","time":{"start":1703038487129,"stop":1703038493570,"duration":6441},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":1,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"testStage":{"description":"Check title of Men's Deals section","status":"skipped","statusMessage":"XFAIL This test is expecting failure == Assertion Error because of error in spelling Men's\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5588093e3d33 <unknown>\n#1 0x5588090a0f87 <unknown>\n#2 0x5588090f0c0e <unknown>\n#3 0x5588090f0d11 <unknown>\n#4 0x55880913c004 <unknown>\n#5 0x55880911a36d <unknown>\n#6 0x558809139297 <unknown>\n#7 0x55880911a0e3 <unknown>\n#8 0x5588090e2044 <unknown>\n#9 0x5588090e344e <unknown>\n#10 0x5588093a8861 <unknown>\n#11 0x5588093ac785 <unknown>\n#12 0x558809396285 <unknown>\n#13 0x5588093ad41f <unknown>\n#14 0x55880937a20f <unknown>\n#15 0x5588093d1028 <unknown>\n#16 0x5588093d11f7 <unknown>\n#17 0x5588093e2ed4 <unknown>\n#18 0x7fb6ad294ac3 <unknown>","statusTrace":"self = <test_sale.TestSalePage.TestSideMenuBarMen object at 0x7f3c099589d0>\nsale_page = <pages.sale_page.SalePage object at 0x7f3c095ff890>\n\n    @allure.title(\"TC 10.02.01 - Verify 'MEN'S DEALS' title is correct\")\n    @pytest.mark.xfail(reason=\"This test is expecting failure == Assertion Error \"\n                              \"because of error in spelling Men's\")\n    def test_tc_10_02_01(self, sale_page):\n        \"\"\"Check title of Men's Deals section\"\"\"\n>       title = sale_page.check_text_in_men_deals_title()\n\ntests/test_sale.py:65: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/sale_page.py:19: in check_text_in_men_deals_title\n    men_deals_title = self.element_is_visible(self.side_bar_locators.MEN_DEALS_TITLE)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"747ce82830cbff6b829f03a4515041ba\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7f3c09b16c00>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5588093e3d33 <unknown>\nE       #1 0x5588090a0f87 <unknown>\nE       #2 0x5588090f0c0e <unknown>\nE       #3 0x5588090f0d11 <unknown>\nE       #4 0x55880913c004 <unknown>\nE       #5 0x55880911a36d <unknown>\nE       #6 0x558809139297 <unknown>\nE       #7 0x55880911a0e3 <unknown>\nE       #8 0x5588090e2044 <unknown>\nE       #9 0x5588090e344e <unknown>\nE       #10 0x5588093a8861 <unknown>\nE       #11 0x5588093ac785 <unknown>\nE       #12 0x558809396285 <unknown>\nE       #13 0x5588093ad41f <unknown>\nE       #14 0x55880937a20f <unknown>\nE       #15 0x5588093d1028 <unknown>\nE       #16 0x5588093d11f7 <unknown>\nE       #17 0x5588093e2ed4 <unknown>\nE       #18 0x7fb6ad294ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException","steps":[{"name":"Check text in Men's deals title","time":{"start":1703038493571,"stop":1703038498658,"duration":5087},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5588093e3d33 <unknown>\n#1 0x5588090a0f87 <unknown>\n#2 0x5588090f0c0e <unknown>\n#3 0x5588090f0d11 <unknown>\n#4 0x55880913c004 <unknown>\n#5 0x55880911a36d <unknown>\n#6 0x558809139297 <unknown>\n#7 0x55880911a0e3 <unknown>\n#8 0x5588090e2044 <unknown>\n#9 0x5588090e344e <unknown>\n#10 0x5588093a8861 <unknown>\n#11 0x5588093ac785 <unknown>\n#12 0x558809396285 <unknown>\n#13 0x5588093ad41f <unknown>\n#14 0x55880937a20f <unknown>\n#15 0x5588093d1028 <unknown>\n#16 0x5588093d11f7 <unknown>\n#17 0x5588093e2ed4 <unknown>\n#18 0x7fb6ad294ac3 <unknown>\n\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/sale_page.py\", line 19, in check_text_in_men_deals_title\n    men_deals_title = self.element_is_visible(self.side_bar_locators.MEN_DEALS_TITLE)\n                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[{"name":"Find a visible element","time":{"start":1703038493571,"stop":1703038498658,"duration":5087},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5588093e3d33 <unknown>\n#1 0x5588090a0f87 <unknown>\n#2 0x5588090f0c0e <unknown>\n#3 0x5588090f0d11 <unknown>\n#4 0x55880913c004 <unknown>\n#5 0x55880911a36d <unknown>\n#6 0x558809139297 <unknown>\n#7 0x55880911a0e3 <unknown>\n#8 0x5588090e2044 <unknown>\n#9 0x5588090e344e <unknown>\n#10 0x5588093a8861 <unknown>\n#11 0x5588093ac785 <unknown>\n#12 0x558809396285 <unknown>\n#13 0x5588093ad41f <unknown>\n#14 0x55880937a20f <unknown>\n#15 0x5588093d1028 <unknown>\n#16 0x5588093d11f7 <unknown>\n#17 0x5588093e2ed4 <unknown>\n#18 0x7fb6ad294ac3 <unknown>\n\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[{"name":"Find a present element","time":{"start":1703038493571,"stop":1703038498658,"duration":5087},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5588093e3d33 <unknown>\n#1 0x5588090a0f87 <unknown>\n#2 0x5588090f0c0e <unknown>\n#3 0x5588090f0d11 <unknown>\n#4 0x55880913c004 <unknown>\n#5 0x55880911a36d <unknown>\n#6 0x558809139297 <unknown>\n#7 0x55880911a0e3 <unknown>\n#8 0x5588090e2044 <unknown>\n#9 0x5588090e344e <unknown>\n#10 0x5588093a8861 <unknown>\n#11 0x5588093ac785 <unknown>\n#12 0x558809396285 <unknown>\n#13 0x5588093ad41f <unknown>\n#14 0x55880937a20f <unknown>\n#15 0x5588093d1028 <unknown>\n#16 0x5588093d11f7 <unknown>\n#17 0x5588093e2ed4 <unknown>\n#18 0x7fb6ad294ac3 <unknown>\n\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[{"name":"locator","value":"('css selector', '.categories-menu > strong:nth-child(3) > span')"},{"name":"timeout","value":"5"}],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[],"parameters":[{"name":"locator","value":"('css selector', '.categories-menu > strong:nth-child(3) > span')"},{"name":"timeout","value":"10"}],"shouldDisplayMessage":false,"stepsCount":1,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":2,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"7308de65ac9c662d","name":"Screenshot","source":"7308de65ac9c662d.png","type":"image/png","size":37744},{"uid":"e501146a7b4f4e07","name":"HTML source","source":"e501146a7b4f4e07.html","type":"text/html","size":786}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":3,"attachmentsCount":2,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"driver::0","time":{"start":1703038498766,"stop":1703038498826,"duration":60},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"clear_allure_result_folder::0","time":{"start":1703039826024,"stop":1703039826024,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"epic","value":"Sale Page"},{"name":"feature","value":"Testing Side Menu Bar - Men's Deals Section"},{"name":"tag","value":"@pytest.mark.xfail(reason='This test is expecting failure == Assertion Error because of error in spelling Men's')"},{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_sale"},{"name":"subSuite","value":"TestSalePage"},{"name":"host","value":"fv-az1208-409"},{"name":"thread","value":"3175-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_sale"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[],"history":{"statistic":{"failed":0,"broken":0,"skipped":4,"passed":0,"unknown":0,"total":4},"items":[{"uid":"5e6f0a8d134acfa1","reportUrl":"https://verafes.github.io/luma_project/163//#testresult/5e6f0a8d134acfa1","status":"skipped","statusDetails":"XFAIL This test is expecting failure == Assertion Error because of error in spelling Men's\n\nAssertionError: Expected title: 'MEN'S DEALS', Actual title: MENS'S DEALS\nassert \"MENS'S DEALS\" == \"MEN'S DEALS\"\n  - MEN'S DEALS\n  + MENS'S DEALS\n  ?    +","time":{"start":1702950065260,"stop":1702950065301,"duration":41}},{"uid":"a8d854c043ba1be","reportUrl":"https://verafes.github.io/luma_project/162//#testresult/a8d854c043ba1be","status":"skipped","statusDetails":"XFAIL This test is expecting failure == Assertion Error because of error in spelling Men's\n\nAssertionError: Expected title: 'MEN'S DEALS', Actual title: MENS'S DEALS\nassert \"MENS'S DEALS\" == \"MEN'S DEALS\"\n  - MEN'S DEALS\n  + MENS'S DEALS\n  ?    +","time":{"start":1702863670928,"stop":1702863670966,"duration":38}},{"uid":"6eecb00251c0929f","reportUrl":"https://verafes.github.io/luma_project/161//#testresult/6eecb00251c0929f","status":"skipped","statusDetails":"XFAIL This test is expecting failure == Assertion Error because of error in spelling Men's\n\nAssertionError: Expected title: 'MEN'S DEALS', Actual title: MENS'S DEALS\nassert \"MENS'S DEALS\" == \"MEN'S DEALS\"\n  - MEN'S DEALS\n  + MENS'S DEALS\n  ?    +","time":{"start":1702776901425,"stop":1702776901470,"duration":45}}]},"tags":["@pytest.mark.xfail(reason='This test is expecting failure == Assertion Error because of error in spelling Men's')"]},"source":"434287bc113e8158.json","parameterValues":[]}